numba/typeinfer.py

```diff
@@ -171,16 +185,17 @@ def __init__(self, target, items, loc):
         self.loc = loc
 
     def __call__(self, typeinfer):
-        typevars = typeinfer.typevars
-        tsets = [typevars[i.name].get() for i in self.items]
-        oset = typevars[self.target]
-        for vals in itertools.product(*tsets):
-            if vals and all(vals[0] == v for v in vals):
-                tup = types.UniTuple(dtype=vals[0], count=len(vals))
-            else:
-                # empty tuples fall here as well
-                tup = types.Tuple(vals)
-            typeinfer.add_type(self.target, tup)
+        with new_error_context("typing of build tuple at {0}", self.loc):
```

Comment:
Not a native English speaker, but "typing of tuple" sounds more correct than "typing of build tuple".